modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy

```diff
@@ -273,6 +275,30 @@ class K8sTaskHandler extends TaskHandler {
         return false
     }
 
+    long getEpochMilli(String timeString) {
+        final time = DateTimeFormatter.ISO_INSTANT.parse(timeString)
+        return Instant.from(time).toEpochMilli()
+    }
+
+    /**
+     * Update task start and end times based on pod timestamps.
+     * We update timestamps because it's possible for a task to run  so quickly
+     * (less than 1 second) that it skips right over the RUNNING status.
+     * If this happens, the startTimeMillis never gets set and remains equal to 0.
+     * To make sure startTimeMillis is non-zero we update it with the pod start time.
+     * We update completTimeMillis from the same pod info to be consistent.
+     */
+    void updateTimestamps(Map terminated) {
+        try {
+            startTimeMillis = getEpochMilli(terminated.startedAt as String)
+            completeTimeMillis = getEpochMilli(terminated.finishedAt as String)
+        } catch( Exception e ) {
+            log.debug "Failed updating timestamps '${terminated.toString()}'", e
+            startTimeMillis = System.currentTimeMillis()
+            completeTimeMillis = System.currentTimeMillis()
+        }
```

Comment:
It would be a bit better to keep in two separate try/catch statements. But ok, I can do.

Reply:
How this sounds to you ? 

```
    void updateTimestamps(Map terminated) {
        try {
            startTimeMillis = getEpochMilli(terminated.startedAt as String)
        } catch( Exception e ) {
            log.debug "Failed updating start timestamp - pod startedAt=${terminated.startedAt}", e
            if( !startTimeMillis )
                startTimeMillis = System.currentTimeMillis()
        }

        try {
            completeTimeMillis = getEpochMilli(terminated.finishedAt as String)
        } catch( Exception e ) {
            log.debug "Failed updating completion timestamp - pod finishedAt=${terminated.finishedAt}", e
            if( !completeTimeMillis )
                completeTimeMillis = System.currentTimeMillis()
        }
    }
```

Reply:
The reason I like one try/catch is to maintain consistency with the timestamps. If you have two try/catch blocks, and `startedAt` fails, but `finishedAt` succeeds, then `startTimeMillis` _could_ have a greater timestamp than `completeTimeMillis`. With just one try/catch, if either fails then we at least know that the timestamps will be consistent.  Make sense?

But I do like that you're checking `if (!startTimeMillis)` in the catch block.  I'll add that to my patch.